tests/test_endpoints_sign_in.py

```diff
@@ -0,0 +1,56 @@
+import os
+
+import pytest
+
+import mintapi
+import mintapi.api
+
+
+USERNAME = os.environ.get("MINTAPI_USERNAME", None)
+PASSWORD = os.environ.get("MINTAPI_PASSWORD", None)
+HEADLESS = os.environ.get("MINTAPI_HEADLESS", True)
+USE_CHROMEDRIVER_ON_PATH = os.environ.get("MINTAPI_CHROMEDRIVER_ON_PATH", False)
+SESSION_PATH = os.environ.get("MINTAPI_SESSION", None)
+MFA_METHOD = os.environ.get("MINTAPI_MFA_METHOD", None)
+MFA_TOKEN = os.environ.get("MINTAPI_MFA_TOKEN", None)
+INTUIT_ACCOUNT = os.environ.get("MINTAPI_INTUIT_ACCOUNT", None)
+pytestmark = pytest.mark.skipif(
+    USERNAME is None or PASSWORD is None,
+    reason="This module requires sign in"
+)
+
+
+@pytest.fixture
+def get_mint_driver() -> mintapi.Mint:
+    mint = mintapi.Mint()
+    mint.driver = mintapi.api._create_web_driver_at_mint_com(
+        HEADLESS,
+        SESSION_PATH,
+        USE_CHROMEDRIVER_ON_PATH,
+    )
+    yield mint
+    mint.close()
+
+
+@pytest.fixture
+def do_sign_in(get_mint_driver: mintapi.Mint) -> mintapi.Mint:
+    mintapi.api._sign_in(
+        USERNAME, PASSWORD, get_mint_driver.driver,
+        MFA_METHOD, MFA_TOKEN, INTUIT_ACCOUNT,
+    )
+    return get_mint_driver
+
+
+def test_sign_in(get_mint_driver: mintapi.Mint):
+    mintapi.api._sign_in(
+        USERNAME, PASSWORD, get_mint_driver.driver,
+        MFA_METHOD, MFA_TOKEN, INTUIT_ACCOUNT,
+    )
+    assert get_mint_driver.driver.current_url.startswith("https://mint.intuit.com/overview.event")
+   
+
+def test_investment_endpoint(do_sign_in: mintapi.Mint):
```

Comment:
I rearranged the sign_in tests to make them a little bit easier to work with. I then added a test for your endpoint. This test fails despite it replicating what your added test on this branch. Do you know why?

Reply:
@jaron-l it looks like you got it to work now.  However, I will note that it looks like you're using an incorrect formatting for declaring a Python method private.  It's my understanding that it requires two underscores, not one.  Just curious if you intended to remove the classification as private?

Reply:
Yeah, I guess I didn't need it anymore. I started by writing tests that call the endpoint functions themselves and so I needed them to be only semi-private so they were testable. They can probably be reverted. I'll fix it. Generally though, I think we should have some sort of definition of what should be public, semi-private, and private as that is a debate in software engineering.

But I don't think you see the issue that I was trying to raise. When I call `__call_investment_endpoint`, it doesn't have he investment key that is required in `get_investment_data`. Now that I think about it, it's probably because I don't have my investments listed in mintapi. We probably need to fail safely from that.

Reply:
Actually, I might want to argue that both `_flatten_utilization` and `_process_utilization` should be semi-private as ideally they would be testable code. I agree that `__call_investment_endpoint` probably can be private.

Reply:
I'm fine on the privacy level mentioned related to utilization. No big deal there. 

And let me know if you still need me to look at the issue you're having.

Reply:
Can you add in the error handling for if the "Investment" key is not found in the return json?

Reply:
Our command line code doesn't actually handle the case that data is None, so you'll need to add that too. #problems-with-duck-typing